File: solutions/cpp/pangram/3/pangram.cpp
#include "pangram.h"
#include <bitset>

namespace pangram {

bool is_pangram(const std::string& input)
{
    std::bitset<26> pangram_map {};
    for (auto ch: input)
    {
        if ('a' <= ch && ch <= 'z')
        {
            pangram_map.set(ch - 'a');
        }
        if ('A' <= ch && ch <= 'Z')
        {
            pangram_map.set(ch - 'A');
        }
    }
    return pangram_map.count() == pangram_map.size();
}

}  // namespace pangram

File: solutions/go/pangram/3/pangram.go
package pangram

import "strings"

func IsPangram(input string) bool {
    if len(input) < 26 {
        return false
    }
	m := make(map[rune]bool)
    for _, v := range strings.ToLower(input) {
        if 'a' <= v && v <= 'z' {
            m[v] = true
        }
    }
    var ok bool
    for r := 'a'; r <= 'z'; r++ {
        if ok = m[r]; !ok {
            return ok
        }
    }
    return ok
}
